src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -68,16 +78,29 @@ public Task<ActionResult> CpuProfile(int? pid, [FromQuery]int duration = 30)
         {
             return InvokeService(async () =>
             {
-                OperationResult<Stream> result = await _diagnosticServices.StartCpuTrace(pid, duration, this.HttpContext.RequestAborted);
-                return CreateFileStreamResult(result, $"{Guid.NewGuid()}.nettrace");
+                OperationResult<IStreamResult> result = await _diagnosticServices.StartCpuTrace(pid, duration, this.HttpContext.RequestAborted);
+                return new EventStreamResult(result.Value, "application/octet-stream", Invariant($"{Guid.NewGuid()}.nettrace"));
             });
         }
 
-        private FileStreamResult CreateFileStreamResult(OperationResult<Stream> result, FormattableString downloadName)
+        [HttpGet("trace/{pid?}")]
+        public Task<ActionResult> Trace(int? pid, [FromQuery]int durationSeconds = 30)
         {
-            //Compression is done automatically by the response
-            //Chunking is done because the result has no content-length
-            return File(result.Value, "application/octet-stream", Invariant(downloadName));
+            return InvokeService(async () =>
+            {
+                OperationResult<IStreamResult> result = await _diagnosticServices.StartTrace(pid, durationSeconds, this.HttpContext.RequestAborted);
+
+                return new EventStreamResult(result.Value, "application/octet-stream", Invariant($"{Guid.NewGuid()}.nettrace"));
+            });
+        }
+
+        [HttpGet("logs/{pid?}")]
+        public ActionResult Logs(int? pid, [FromQuery] int durationSeconds = 30)
+        {
+            return new OutputStreamResult( async (outputStream, token) =>
+            {
+                await _diagnosticServices.StartLogs(outputStream, pid, durationSeconds, token);
+            }, "application/x-ndjson", Invariant($"{Guid.NewGuid()}.txt"));
```

Comment:
Note this will probably change to default to text/event-stream, or at the very least accept both types.

Reply:
Based on meeting: support a curl like scenario, so text data should be fine. We can do event-stream based on user agent, and respect headers for the content type.